apps/passport-client/src/localstorage.ts

```diff
@@ -72,3 +72,11 @@ export function loadIdentity(): Identity | null {
 export function saveIdentity(identity: Identity): void {
   window.localStorage["identity"] = identity.toString();
 }
+
+export function saveLastSyncedIds(id: string) {
+  localStorage["last-synced-ids"] = id;
+}
+
+export function loadLastSyncedIds(): string {
+  return localStorage["last-synced-ids"];
```

Comment:
I'm not sure need localStorage for this.

If we're loading from the server every time the passport loads anyway, then you can store the PCD IDs in state.

I recommend something like the following

...in ZuState:
```ts
lastSyncedIds?: string // instead of hasUploaded...
```
then, in trySync:
- if `!lastSyncedIds` , download
- otherwise if `lastSyncedIds !== currentIds`, upload

Reply:
the less we have in localStorage, the better IMO---less scope for bugs as new versions of the app see localStorage values written by older versions

Reply:
I understand the impulse to keep `localStorage` as simple as possible. However, I don't think your suggestion works.

Consider the case when a PCD was just added to the passport. 

The 'Add PCD' popup isn't the screen which uploads the PCD, it closes immediately. The newly added PCD must be uploaded the next time the logged in user loads the passport home screen, by the Home Screen. At the moment in time that the home screen is loaded after a PCD was added, the passport needs to decide: is it uploading, or is it downloading. I don't think this is possible to do properly without something in `localStorage`.

To go one level deeper, the passport needs to make this decision, because neither the policy 'always download when the homescreen opens' nor the policy 'always upload when the homescreen opens' works.

If we 'always download' - newly added PCDs never get uploaded, and get overwritten by the server state.

If we 'always upload' - the passport will immediately overwrite server-state.

Reply:
The Add PCD popup has to upload before it closes. I can show that RippleLoader till then 

If a new PCD can be in a kind of pending state --- added locally, but not uploaded, for an indefinite time till the next time the user opens the passport -- we'll get bad overwriting issues no matter what.

For example:
- On laptop: user adds a new card using the Eth PCD screen. Add PCD popup completes successfully, page says "Done", user thinks they are done.
- On phone: user adds two new cards using other apps from this weekend's hackathon.
- Finally, user opens Zupass on their laptop The two new cards are now clobbered.

Reply:
I think this may let us remove the `syncedIds` concatenation entirely, both from localStorage and from ZuState.

The rule would be something like this:
- Whenever the passport opens, download
- In the Add PCD screen, upload
- Immediately user deletes a PCD, upload

...this also makes it easy to add a fourth case:
- Every 5 minutes (or, if we want to be fancy, immediately using the Web Push API), upload